src/JWTGuard.php

```diff
@@ -74,9 +74,14 @@ public function user()
 
         if ($this->jwt->getToken() && $this->jwt->check()) {
             $id = $this->jwt->payload()->get('sub');
+            $provider = $this->jwt->payload()->get('prv');
 
-            return $this->user = $this->provider->retrieveById($id);
+            if ($this->provider->getModel() == $provider) {
+                $this->user = $this->provider->retrieveById($id);
```

Comment:
No need to go further, you can just `return $this->user = $this->provider->retrieveById($id);`
as the final `return $this->user;` (`null`) is useless